### ISSUE TITLE
Cannot tab on to or past a web-component in a container

### ISSUE THREAD

**Issue by obambrough** [focus-trap/focus-trap#643]
Add a focus trap on a container that has a web-component as a child element. Focus cannot be moved to or passed the web-component element, the focus gets stuck on the element before the web-component.

Here's a sandbox showing the issue: https://codesandbox.io/s/agitated-murdock-6q9kks
NOTE: When loading the sandbox I don't know why it doesn't run automatically - press the reload icon in the small codesandbox.io browser to see it.

I believe this was introduced as part of the PR https://github.com/focus-trap/focus-trap/pull/516

**Comment by stefcameron** [comment]
Thanks for reporting this, @obambrough ! I'm not entirely sure #496 is the culprit since your use case may not have been considered back then. IIRC, #496 was about supporting a trap _inside_ a shadow dom (web component). In your case, you have a web component inside the trap and you're expecting (rightly so) to be able to tab to that button just like the other buttons.

Shadow DOM support is a big change. We're actually working on that in tabbable (since that's where it has to come from): https://github.com/focus-trap/tabbable/pull/570

I have published a beta of tabbable and also a beta of focus-trap (`v6.8.0-beta.1` so far). Unfortunately, it doesn't work with that either, and I know why: Because when you try to tab to the button inside the web component, even with the trap enabling shadow DOM (with the new tabbable beta) and finding the button in the first place, focus-trap is using the `Node.contains()` API to determine if focus has escaped the trap when you hit the tab key, and `contains()` doesn't look inside web components (open or closed). So it thinks focus has escaped, and pulls it back in by setting focus to the most-recently-focused node, which is the one either just before (forward tab) or after (reverse tab) the web component's button.

So all that to say, the tabbable beta is necessary to fix this, but then focus-trap still has some work to do for shadow DOM support even with tabbable finding elements inside web components.

**Comment by stefcameron** [comment]
@obambrough This should now be fixed in `focus-trap@6.8.0-beta.2` provided you configure your trap with the new `tabbableOptions: { getShadowRoot: true }` option.

This was the fix, if you're curious, over on #610: https://github.com/focus-trap/focus-trap/pull/610/commits/5c6b5f3b8bb6d3c13074f7525d49d7a8a26ca46f

![with-open-web-component](https://user-images.githubusercontent.com/2855350/158031822-5c85d3a1-e8e7-466e-bee5-91c3fdc8fe14.gif)

**Comment by obambrough** [comment]
Awesome, thank you!

**Comment by stefcameron** [comment]
My pleasure!

**Comment by stefcameron** [comment]
focus-trap v6.8.0 with this fix is being published now.

**Comment by obambrough** [comment]
Really appreciate the notice that its in a release, thank you.
